### PR TITLE
[dagster-dbt] [software-defined-assets] update to reference new implementation

### PR DESCRIPTION
**Pull request by OwenKephart**
relies on: https://github.com/dagster-io/dagster/pull/6487

### CODE REVIEW

File: python_modules/libraries/dagster-dbt/dagster_dbt/asset_defs.py [hunk 1]
```diff
@@ -35,42 +36,72 @@ def _get_node_name(node_info: Mapping[str, Any]):
...
-    description = "\n\n".join(filter(None, description_sections))
-
-    @asset(
-        name=node_info["name"],
-        description=description,
-        non_argument_deps={
-            AssetKey(dep_name.split(".")[-1]) for dep_name in node_info["depends_on"]["nodes"]
-        },
+) -> AssetsDefinition:
+    outs = {}
```
Comment (sryza) [inline python_modules/libraries/dagster-dbt/dagster_dbt/asset_defs.py:46]: Type annotations on these collections would aid readability

**Review by sryza — APPROVED** [review]
Nice

File: python_modules/libraries/dagster-dbt/dagster_dbt/asset_defs.py [hunk 2]
```diff
@@ -35,42 +36,72 @@ def _get_node_name(node_info: Mapping[str, Any]):
...
-        context.resources.dbt.run(models=[".".join([node_info["package_name"], node_info["name"]])])
-        if runtime_metadata_fn:
-            metadata = runtime_metadata_fn(context, node_info)
-            yield Output(None, metadata=metadata)
-        else:
-            yield Output(None)
-
-    return _node_asset
+    def _dbt_project_multi_assset(context):
+        dbt_output = context.resources.dbt.run()
```
Comment (amarrella) [inline python_modules/libraries/dagster-dbt/dagster_dbt/asset_defs.py:87]: Hi @OwenKephart , following up on https://github.com/dagster-io/dagster/pull/6309#discussion_r802945954 
To make it possible to keep the `models` parameter here so the dbt graph can be split into multiple parts? Use case is still the one highlighted in the comment linked: being able to interleave dbt and python computation (e.g. a machine learning job that runs in between table transformations)

Comment (OwenKephart) [reply python_modules/libraries/dagster-dbt/dagster_dbt/asset_defs.py:87]: hi @amarrella ! the load_assets_from_dbt_project() function accepts the "select" argument, which will allow you to filter for subsets of models (as far as I know, --select is basically equivalent to --models in newer versions of dbt).

However, on closer inspection, this argument doesn't quite work as expected at the moment, but it is a feature that we'll have working for the 0.14.0 release.

**Pull request merged**